src/MagnumPlugins/BasisImporter/Test/BasisImporterTest.cpp

```diff
@@ -0,0 +1,188 @@
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNETCION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include <sstream>
+#include <Corrade/Containers/Optional.h>
+#include <Corrade/TestSuite/Tester.h>
+#include <Corrade/TestSuite/Compare/Container.h>
+#include <Corrade/Utility/Directory.h>
+#include <Corrade/Utility/DebugStl.h>
+#include <Corrade/Utility/FormatStl.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Trade/AbstractImporter.h>
+#include <Magnum/Trade/ImageData.h>
+
+#include "MagnumPlugins/BasisImporter/BasisImporter.h"
+
+#include "configure.h"
+
+namespace Magnum { namespace Trade { namespace Test {
+
+struct BasisImporterTest: TestSuite::Tester {
+    explicit BasisImporterTest();
+
+    void empty();
+    void invalid();
+
+    void openTwice();
+    void importTwice();
+
+    void configValue();
+
+    void rgb();
+    void rgba();
+
+    /* Explicitly forbid system-wide plugin dependencies */
+    PluginManager::Manager<AbstractImporter> _manager{"nonexistent"};
+};
+
+BasisImporterTest::BasisImporterTest() {
+    addTests({&BasisImporterTest::empty,
+              &BasisImporterTest::invalid,
+
+              &BasisImporterTest::openTwice,
+              &BasisImporterTest::importTwice,
+
+              &BasisImporterTest::configValue});
+
+    const Int numTargetFormats =
+        Int(BasisTranscodingType::Pvrtc1_4OpaqueOnly)+1;
+    addInstancedTests({&BasisImporterTest::rgb,
+            &BasisImporterTest::rgba}, numTargetFormats);
+
+    /* Load the plugin directly from the build tree. Otherwise it's static and
+       already loaded. */
+    #ifdef BASISIMPORTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_manager.load(BASISIMPORTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+}
+
+void BasisImporterTest::empty() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    std::ostringstream out;
+    Error redirectError{&out};
+    char a{};
+    /* Explicitly checking non-null but empty view */
+    CORRADE_VERIFY(!importer->openData({&a, 0}));
+    CORRADE_COMPARE(out.str(), "Trade::BasisImporter::openData(): the file is empty\n");
+}
+
+void BasisImporterTest::invalid() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!importer->openFile(Utility::Directory::join(PNGIMPORTER_TEST_DIR, "rgb.png")));
```

Comment:
Could be just `openData("hello?")`, you wouldn't need to bother with PNGIMPORTER_TEST_DIR or anything else.

Reply:
Yes, definitely.